sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/ActionDescriptorFactory.scala

```diff
@@ -75,24 +76,58 @@ private[impl] object ActionDescriptorFactory extends ComponentDescriptorFactory
+    def combineByES(subscriptions: Seq[KalixMethod]): Seq[KalixMethod] = {
+      def groupByES(methods: Seq[KalixMethod]): Map[String, Seq[KalixMethod]] = {
+        val withEventSourcedIn = methods.filter(kalixMethod =>
+          kalixMethod.methodOptions.exists(option =>
+            option.hasEventing && option.getEventing.hasIn && option.getEventing.getIn.hasEventSourcedEntity))
+        //Assuming there is only one topic annotation per method, therefore head is as good as any other
+        withEventSourcedIn.groupBy(m => m.methodOptions.head.getEventing.getIn.getEventSourcedEntity)
+      }
+      groupByES(subscriptions).collect {
+        case (eventSourcedEntity, kMethods) if kMethods.size > 1 =>
+          val methodsMap: Map[String, Method] = kMethods.map { k =>
+            (k.serviceMethod.javaMethodOpt.get.getParameterTypes()(0).getName, k.serviceMethod.javaMethodOpt.get)
+          }.toMap
+
+          KalixMethod(
+            RestServiceMethod(
+              kMethods.head.serviceMethod.javaMethodOpt.get, //TODO once we have 'methodsMap' this won't be used thus .head should be find. Would it make sense use set this field to OptionFIXME apart from .head would it make sense use set this field to Option?
+              methodName = "KalixSyntheticMethodOnES" + eventSourcedEntity.capitalize,
+              methodsMap = methodsMap))
+            .withKalixOptions(kMethods.head.methodOptions)
+        case (eventSourcedEntity, kMethod +: Nil) =>
+          kMethod
+      }.toSeq
+    }
+
+    def combineByTopic(subscriptions: Seq[KalixMethod]): Seq[KalixMethod] = {
       def groupByTopic(methods: Seq[KalixMethod]): Map[String, Seq[KalixMethod]] = {
         val withTopicIn = methods.filter(kalixMethod =>
           kalixMethod.methodOptions.exists(option =>
             option.hasEventing && option.getEventing.hasIn && option.getEventing.getIn.hasTopic))
-        //Assuming there is only one topic, therefore head is as good as any other
+        //Assuming there is only one topic annotation per method, therefore head is as good as any other
         withTopicIn.groupBy(m => m.methodOptions.head.getEventing.getIn.getTopic)
       }
       groupByTopic(subscriptions).collect {
         case (topic, kMethods) if kMethods.size > 1 =>
-          throw InvalidComponentException(
-            s"topic: '$topic' it is used in multiple @Subscription.Topic annotations. Each @Subscription.Topic must point to a different topic")
-      }
-      subscriptions
+          val methodsMap: Map[String, Method] = kMethods.map { k =>
+            (k.serviceMethod.javaMethodOpt.get.getParameterTypes()(0).getName, k.serviceMethod.javaMethodOpt.get)
+          }.toMap
+
+          KalixMethod(
+            RestServiceMethod(
+              kMethods.head.serviceMethod.javaMethodOpt.get, //TODO once we have 'methodsMap' this won't be used thus .head should be find. Would it make sense use set this field to Option?
```

Comment:
yes, indeed. I think we should remove it. See my comments further down.

Reply:
I didn't remove it. Please have a look at the new Model and let me know